lib/ruby_indexer/lib/ruby_indexer/declaration_listener.rb

```diff
@@ -314,20 +314,22 @@ def on_def_node_enter(node)
           @owner_stack.last,
         ))
       when Prism::SelfNode
-        singleton = singleton_klass
+        owner = @owner_stack.last
 
-        @index.add(Entry::Method.new(
-          method_name,
-          @file_path,
-          node.location,
-          node.name_loc,
-          comments,
-          list_params(node.parameters),
-          current_visibility,
-          singleton,
-        ))
+        if owner
+          singleton = @index.existing_or_new_singleton_class(owner.name)
+
+          @index.add(Entry::Method.new(
```

Comment:
That becomes a bit messy because the case statement helps us ignore dynamic definitions like `def foo.bar`. However, we also allow `owner` to be `nil` at the moment, which means we can't use that as the check for pushing a new entry into the index.

In the end, we clear the case statement, but then get a bunch of duplicate if statements that are already checked inside the case just to avoid pushing incorrect entries.

I think this will become easier once we turn `owner` into non nilable.